merge_request_approvals.go

```diff
@@ -76,6 +76,30 @@ type MergeRequestApprovalRule struct {
 	ContainsHiddenGroups bool                 `json:"contains_hidden_groups"`
 }
 
+// MergeRequestApprovalState represents a GitLab merge request approval state.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/merge_request_approvals.html#get-the-approval-state-of-merge-requests
+type MergeRequestApprovalState struct {
+	ApprovalRulesOverwritten bool                             `json:"approval_rules_overwritten"`
+	Rules                    []*MergeRequestApprovalStateRule `json:"rules"`
+}
+
+// MergeRequestApprovalStateRule represent rules as part of the MergeRequestApprovalState
+type MergeRequestApprovalStateRule struct {
+	ID                   int                  `json:"id"`
+	Name                 string               `json:"name"`
+	RuleType             string               `json:"rule_type"`
+	EligibleApprovers    []*BasicUser         `json:"eligible_approvers"`
+	ApprovalsRequired    int                  `json:"approvals_required"`
+	SourceRule           *ProjectApprovalRule `json:"source_rule"`
+	Users                []*BasicUser         `json:"users"`
+	Groups               []*Group             `json:"groups"`
+	ContainsHiddenGroups bool                 `json:"contains_hidden_groups"`
+	ApprovedBy           []*BasicUser         `json:"approved_by"`
+	Approved             bool                 `json:"approved"`
+}
```

Comment:
Let's not create a new struct, but instead add the missing fields to the `MergeRequestApprovalRule` struct and reuse that one here.

Reply:
Done